### PR TITLE
Inline functions when loading the dictionary

### PR DESCRIPTION
**Pull request by HuwCampbell**
The functions sourceCheck and sourceInline are more or less taken from Repl, so should probably be unified somewhere else.

This makes the virtual definitions the dictionary produces be "complete", in that they don't need any more actions performed on them before they be evaluated with the source evaluator or compiled to core.

### CODE REVIEW

File: src/Icicle/Storage/Dictionary/Toml.hs [hunk 1]
```diff
@@ -102,20 +105,19 @@ loadDictionary' parentFuncs parentConf parentConcrete fp
...
-       )
+    <- (\(a, q) ->
+          (<$>)
+         (\(q', _) -> DictionaryEntry a (VirtualDefinition (Virtual q')))
+        $ (=<<)
+         (\q' -> sourceCheck d'' q' )
+        $ (<$>)
+         (\(q', _) -> sourceInline d' q')
+         ( sourceCheck d'' q )
+       ) `traverse` virtualDefinitions'
```
Comment (amosr) [inline src/Icicle/Storage/Dictionary/Toml.hs]: I think this would be clearer with explicit do notation or something

Comment (tranma) [inline src/Icicle/Storage/Dictionary/Toml.hs]: @HuwCampbell plays master-level golf

Comment (domdere) [inline src/Icicle/Storage/Dictionary/Toml.hs]: lol

Comment (amosr) [inline src/Icicle/Storage/Dictionary/Toml.hs]: not in MY backyard

Comment (HuwCampbell) [inline src/Icicle/Storage/Dictionary/Toml.hs]: Ok, ok.

**Comment by amosr** [comment]
groovy :purple_heart:

**Pull request merged**